doc/src/tutorials/arch/heterogeneous_tiles.rst

```diff
@@ -0,0 +1,158 @@
+                <input name="CE0" num_pins="1"/>
+                <input name="CE1" num_pins="1"/>
+                <input name="IGNORE0" num_pins="1"/>
+                <input name="IGNORE1" num_pins="1"/>
+                <input name="S0" num_pins="1"/>
+                <input name="S1" num_pins="1"/>
+                <output name="O" num_pins="1"/>
+                <fc in_type="abs" in_val="2" out_type="abs" out_val="2"/>
+                <pinlocations pattern="custom">
+                    <loc side="right">BUFGCTRL_1.S1 BUFGCTRL_1.CLK_BUFG_I0 BUFGCTRL_1.CE1 BUFGCTRL_1.CLK_BUFG_I1 BUFGCTRL_1.IGNORE1 BUFGCTRL_1.IGNORE0 BUFGCTRL_1.CE0 BUFGCTRL_1.S0</loc>
+                    <loc side="left">BUFGCTRL_1.CLK_BUFG_I0 BUFGCTRL_1.CLK_BUFG_I1 BUFGCTRL_1.CLK_BUFG_O</loc>
+                </pinlocations>
+                <equivalent_sites>
+                  <site pb_type="BUFGCTRL" pin_mapping="direct"/>
+                </equivalent_sites>
+            </sub_tile>
+        </tile>
+    </tiles>
+
+    <complexblocklist>
+        <pb_type name="BUFGCTRL"/>
+            <clock name="I0" num_pins="1"/>
+            <clock name="I1" num_pins="1"/>
+            <input name="CE0" num_pins="1"/>
+            <input name="CE1" num_pins="1"/>
+            <input name="IGNORE0" num_pins="1"/>
+            <input name="IGNORE1" num_pins="1"/>
+            <input name="S0" num_pins="1"/>
+            <input name="S1" num_pins="1"/>
+            <output name="O" num_pins="1"/>
+            <mode ...>
+            ...
+        </pb_type>
+    </complexblocklist>
+
+2. Different sub tile types.
+
+   As another example taken from the Xilinx Series 7 fabric, the HCLK_IOI tile is composed of three different site types, namely BUFIO, BUFR and IDELAYCTRL.
+
+   Each one of these site types has different IO pins as well as pin locations.
+
+   A visual representation of this case is the following:
+
+.. image:: hclk_ioi.png
+
+.. code-block:: xml
+
+    <tile name="HCLK_IOI">
+        <sub_tile name="BUFIO" capacity="4">
+            <clock name="I" num_pins="1"/>
+            <output name="O" num_pins = "1"/>
+            <equivalent_sites>
+                <site name="BUFIO_SITE" pin_mapping="direct"/>
+            </equivalent_sites>
+            <fc ...>
+            <pinlocations ...>
+        </sub_tile>
+        <sub_tile name="BUFR" capacity="4">
+            <clock name="I" num_pins="1"/>
+            <input name="CE" num_pins="1"/>
+            <output name="O" num_pins = "1"/>
+            <equivalent_sites>
+                <site name="BUFR_SITE" pin_mapping="direct"/>
+            </equivalent_sites>
+            <fc ...>
+            <pinlocations ...>
+        </sub_tile>
+        <sub_tile name="IDELAYCTRL" capacity="1">
+            <clock name="REFCLK" num_pins="1"/>
+            <output name="RDY" num_pins="1"/>
+            <equivalent_sites>
+                <site name="IDELAYCTRL_SITE" pin_mapping="direct"/>
```

Comment:
This example uses `<site name=` while the previous example uses `<site pb_type=` we should make sure they are consistent with the actual XML format.